pygments/lexers/dotnet.py

```diff
@@ -684,3 +694,18 @@ class FSharpLexer(RegexLexer):
             (r'"', String),
         ],
     }
+
+    def analyse_text(text):
+        """F# doesn't have that many unique features -- |> for matching
+        seems quite common though, in addition to let/match."""
```

Comment:
let/match/-> are quite common in functional and derived languages, e.g. Haskell or Rust.  I wouldn't use them.